File: angular-movieFrontend/src/Data/Response/MovieResponse.ts
import { DirectorResponse } from "./DirectorResponse";
import { GenreResponse } from "./GenreResponse";
import { MediaStatsResponse } from "./MediaStatsResponse";
import { ReviewResponse } from "./ReviewResponse";

export interface MovieResponse{
  id: string;
  title: string;
  description: string;
  genre: GenreResponse;
  director: DirectorResponse;
  releaseDate: string;
  language: string | null;
  reviews: ReviewResponse[] | null;
  mediaStats: MediaStatsResponse;
  duration: string;
  distributionType: string;
  status: string;
}

File: Application/Common/DTO/Response/MovieResponse.cs
namespace Application.Common.DTO.Response
{
    public record MovieResponse(
        Guid id,
        string Title,
        string Description,
        GenreResponse Genre,
        DirectorResponse Director,
        DateTime ReleaseDate,
        string? Language,
        List<ReviewResponse>? Reviews,
        MediaStatsResponse MediaStats,
        TimeSpan Duration,
        bool IsCinemaRelease

        ) : MediaResponse(id, Title, Description, Genre, ReleaseDate, Language, Reviews), IResponse;

}
